include/flatbuffers/util.h

```diff
@@ -321,7 +321,10 @@ inline int FromUTF8(const char **in) {
       break;
     }
   }
-  if ((**in << len) & 0x80) return -1;  // Bit after leading 1's must be 0.
+  // the input byte can be negative.
+  // '<<' of a negative value is undefined behavior
+  const unsigned char unsignedIn = static_cast<unsigned char>(**in);
```

Comment:
The surrounding code does not use camerCase. Actually just put the cast in-line, and no need for the comment.